### PR TITLE
Update Nuvoton Ethernet project file

### PR DESCRIPTION
**Pull request by qiutongs**
Description
-----------

Update Nuvoton Ethernet project file

Enable ipconfigUSE_TCP_WIN to fix a link error in this line: https://github.com/aws/amazon-freertos/blame/master/libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_TCP_IP.c#L1326

Without ipconfigUSE_TCP_WIN == 1, ulTCPWindowTxSack is undefined

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by htibosch** [comment]
Hello @qiutongs : oops, that was not the intention.
When `ipconfigUSE_TCP_WIN` == 0, `ulTCPWindowTxSack` should not be called.
Are you compiling with today's master branch?

( for those who wonder: `ipconfigUSE_TCP_WIN` allows TCP to use sliding windows. `ulTCPWindowTxSack()` has to do with SACK: Selective Acknowledgements. That is a TCP option which is only used when TCP sliding windows are being used ).

**Comment by cobusve** [comment]
Looking in the code on the master branch it is always calling ulTCPWindowTxSack from prvSkipPastRemainingOptions, so it seems like we need some logic here to avoid this call when not using sliding windows?

Hein can you please advise what the appropriate fix would be here?

**Comment by qiutongs** [comment]
> Hello @qiutongs : oops, that was not the intention.
> When `ipconfigUSE_TCP_WIN` == 0, `ulTCPWindowTxSack` should not be called.
> Are you compiling with today's master branch?
> 
> ( for those who wonder: `ipconfigUSE_TCP_WIN` allows TCP to use sliding windows. `ulTCPWindowTxSack()` has to do with SACK: Selective Acknowledgements. That is a TCP option which is only used when TCP sliding windows are being used ).

Yes, I am compiling the latest master. 

When `ipconfigUSE_TCP_WIN` == 0, should `ulTCPWindowTxSack` be compiled? The answer is no in current code: https://github.com/aws/amazon-freertos/blob/master/libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_TCP_WIN.c#L1676

So I think ulTCPWindowTxSack and prvSkipPastRemainingOptions should be both compiled or both not compiled. Please share your feedback.

File: vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/FreeRTOSIPConfig.h [hunk 1]
```diff
@@ -242,7 +242,7 @@ extern uint32_t numaker_ulRand(void);
 #define ipconfigUSE_TCP                                ( 1 )
 
 /* USE_WIN: Let TCP use windowing mechanism. */
-#define ipconfigUSE_TCP_WIN                            ( 0 )
+#define ipconfigUSE_TCP_WIN                            ( 1 )
```
Comment (aggarw13) [inline vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/FreeRTOSIPConfig.h:245]: How does the build pass for other boards (ST, TI, NXP ) that also do not have this macro set?

Comment (aggarw13) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/FreeRTOSIPConfig.h:245]: I looked at the call-sites for `ulTCPWindowTxSack`, and I see only a single call from [`prvReadSackOption`](https://github.com/aws/amazon-freertos/blob/master/libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_TCP_IP.c#L1336) that is currently protected by [`ipconfigUSE_TCP_WIN`](https://github.com/aws/amazon-freertos/blob/master/libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_TCP_IP.c#L1302-L1322) 
Does the problem not exist anymore?

Comment (aggarw13) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/FreeRTOSIPConfig.h:245]: It seems that `prvSkipPastRemainingOptions` was removed in #2015 by @htibosch

Comment (qiutongs) [reply vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/config_files/FreeRTOSIPConfig.h:245]: Yeah, I will rebase to the latest master and check if Nuvoton eth project can build.

**Comment by qiutongs** [comment]
Temporarily closed this PR.

**Comment by htibosch** [comment]
Hello, as you are working on the Nuveton project, maybe you have any idea why PR #2029  is getting a build error when building the Nuveton project. Thank you!

**Pull request closed without merging**